### DISCUSSION TITLE
How to contribute to debug console context menu

### DISCUSSION THREAD

**Discussion by yuqizhou77** [microsoft/vscode-discussions#2422] [Extension Development QnA] [answered]
I am hoping to add a command when user view the debug console. Like if user right click on the debug console, an extension could allow user to trigger a command from
![image](https://github.com/user-attachments/assets/7de482ca-9c58-44b4-b78d-1bf52570e410)

Can we add own command to the menu?

**Comment by alefragnani** [comment, marked as answer]
Hi @yuqizhou77 ,

I’m not too familiarized with the Debug Adapter API, but based its documentation (https://code.visualstudio.com/api/extension-guides/debugger-extension), it seems you can only interact with the content of the Debug Console window, _for interactive evaluation with autocomplete_. Also, the Contributions Points for menus (https://code.visualstudio.com/api/references/contribution-points#contributes.menus) doesn’t expose any option that refers to the Debug Console.

So, I suppose such feature is not supported. In this case, I would suggest you to open a feature request in the VS Code repo (https://github.com/microsoft/vscode/issues).

Hope this helps
